app/packs/src/components/models/Sample.js

```diff
@@ -319,11 +321,29 @@ export default class Sample extends Element {
       molecular_mass: this.molecular_mass,
       sum_formula: this.sum_formula,
       segments: this.segments.map(s => s.serialize()),
+      color: this.color,
+      state: this.state,
```

Comment:
May be confused with the component state of react. Better find another name (e.g. aggregateState).